### PR TITLE
Adding instructions to setup cs with wsl2

### PR DESCRIPTION
**Pull request by adelSalah78**
Adding instructions to setup cs with wsl2

### CODE REVIEW

**Review by strieflin — CHANGES_REQUESTED** [review]
Hi @adelSalah78, thank you very much for opening this PR. The content you want to have added, however, should not be part of the README of this repo but of the _content of the website_, which you can find in the `docs` folder. In addition, many of the steps you describe are already covered by https://carbynestack.io/getting-started/prerequisites/. You should only describe steps specific to WSL. Depending on how big the differences are it might be best to rewrite https://carbynestack.io/getting-started/prerequisites/ to contain WSL and Linux specific instructions. If the steps are the same and we only need a couple of steps to be done to get a Linux via WSL and can then continue with the steps for Linux already describe, then adding a new page with the WSL-specific setup steps might make more sense.

**Comment by adelSalah78** [comment]
> Hi @adelSalah78, thank you very much for opening this PR. The content you want to have added, however, should not be part of the README of this repo but of the _content of the website_, which you can find in the `docs` folder. In addition, many of the steps you describe are already covered by https://carbynestack.io/getting-started/prerequisites/. You should only describe steps specific to WSL. Depending on how big the differences are it might be best to rewrite https://carbynestack.io/getting-started/prerequisites/ to contain WSL and Linux specific instructions. If the steps are the same and we only need a couple of steps to be done to get a Linux via WSL and can then continue with the steps for Linux already describe, then adding a new page with the WSL-specific setup steps might make more sense.

Hi @strieflin , Kindly find the updates that I have made into the documentations of setting up CS with WSL2 on Windows.
Please let me know if there are any updates needed.

> Hi @adelSalah78, thank you very much for opening this PR. The content you want to have added, however, should not be part of the README of this repo but of the _content of the website_, which you can find in the `docs` folder. In addition, many of the steps you describe are already covered by https://carbynestack.io/getting-started/prerequisites/. You should only describe steps specific to WSL. Depending on how big the differences are it might be best to rewrite https://carbynestack.io/getting-started/prerequisites/ to contain WSL and Linux specific instructions. If the steps are the same and we only need a couple of steps to be done to get a Linux via WSL and can then continue with the steps for Linux already describe, then adding a new page with the WSL-specific setup steps might make more sense.

File: docs/getting-started/prerequisites-wsl2.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
+# Setting up the prerequisites with WSL2
+
+This guide describes how to install the prerequisites required to deploy
+Carbyne Stack via WSL2 in windows.
```
Comment (strieflin) [inline docs/getting-started/prerequisites-wsl2.md:4]: **W**indows

File: docs/getting-started/prerequisites-wsl2.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+!!! warning
+    Carbyne Stack has been tested using the **exact versions** of the tools
+    specified below. Deviating from this _battle tested_ configuration may
+    create all kinds of issues.
+
+## Prerequisites
+
+    WSL2 must be installed in your Windows machine.
```
Comment (strieflin) [inline docs/getting-started/prerequisites-wsl2.md:13]: *o*n

File: docs/getting-started/prerequisites-wsl2.md [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+    To install WSL2, run your Windows powershell and run this command
```
Comment (strieflin) [inline docs/getting-started/prerequisites-wsl2.md:15]: *P*owershell

File: docs/getting-started/prerequisites-wsl2.md [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+
+    ```shell
+    wsl --install
+    ```
+
+Or you can refer to https://www.windowscentral.com/how-install-wsl2-windows-10 if more details are needed.
```
Comment (strieflin) [inline docs/getting-started/prerequisites-wsl2.md:21]: Don't use plain links as they reduce readability:

> Or you can refer to [this guide](https://www.windowscentral.com/how-install-wsl2-windows-10) if more details are needed.

File: docs/getting-started/prerequisites-wsl2.md [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+
+## Platform Setup Prerequisites
+
+Software to be installed:
+
+### install Ubuntu-20.04 inside WSL2 and connect to it
```
Comment (strieflin) [inline docs/getting-started/prerequisites-wsl2.md:27]: **I**nstall

Comment (strieflin) [inline docs/getting-started/prerequisites-wsl2.md:27]: Use "Ubuntu 20.04" w/o the hyphen, please.

File: docs/getting-started/prerequisites-wsl2.md [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+    ```shell
+    wsl install Ubuntu-20.04\
+    wsl -d Ubuntu-20.04
+    ```
+
+### Configure DNS Resolution
+    1. prevent creating the resolve.conf on startup
```
Comment (strieflin) [inline docs/getting-started/prerequisites-wsl2.md:34]: Always use capital letters at the beginning of sentences.

File: docs/getting-started/prerequisites-wsl2.md [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+    ```shell
+    sudo nano /etc/wsl.conf
+    ```
+    And add this text to wsl.conf file
```
Comment (strieflin) [inline docs/getting-started/prerequisites-wsl2.md:38]: Use

> And add this text to `wsl.conf` file

File: docs/getting-started/prerequisites-wsl2.md [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+
+    ```shell
+    [network]
+    generateResolvConf = false
+    ```
+
+    then press CTRL+X to save your file
```
Comment (strieflin) [inline docs/getting-started/prerequisites-wsl2.md:45]: Well, I think that is too much detail. People should know how to save in their editor 😀

File: docs/getting-started/prerequisites-wsl2.md (hunk 7 above)
Comment (strieflin) [reply docs/getting-started/prerequisites-wsl2.md:38]: Please check that you adhere to this style for filenames throughout the whole file

File: docs/getting-started/prerequisites-wsl2.md [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+
+    2. Restart WSL via PowerShell
+    ```shell
```
Comment (strieflin) [inline docs/getting-started/prerequisites-wsl2.md:48]: Add an empty line between text and code snippets. Make sure to adhere to this rule throughout the whole file.

**Comment by strieflin** [comment]
Not quite there yet @adelSalah78 😉:

1. Please check the PR regarding consistent  style, grammar, and spelling.
2. Please make sure that the guide doesn't repeat the steps described [here](https://carbynestack.io/getting-started/prerequisites/). I'd suggest to remove all duplicate steps from your PR.
3. The links to the other guides at the end are not necessary, as a link to the next page is rendered by `mkdocs` anyway.
4. Please add a step (new step 2) to the enumeration at the end of https://carbynestack.io/getting-started/ that links to your guide.

**Review by strieflin — CHANGES_REQUESTED** [review]

**Comment by strieflin** [comment]
Closed due to inactivity. Feel free to reopen when requested changes have been made.

**Pull request closed without merging**